loky/process_executor.py

```diff
@@ -1076,16 +1087,29 @@ def _adjust_process_count(self):
                     self._initargs, self._processes_management_lock,
                     self._timeout, worker_exit_lock, _CURRENT_DEPTH + 1)
             worker_exit_lock.acquire()
+
+            worker_id = -1
+            if _CURRENT_DEPTH == 0:
+                used_ids = set(self._process_worker_ids.values())
+                available_ids = set(range(self._max_workers)) - used_ids
+                if len(available_ids):
+                    worker_id = available_ids.pop()
+
             try:
                 # Try to spawn the process with some environment variable to
                 # overwrite but it only works with the loky context for now.
+                env = self._env
+                if _CURRENT_DEPTH == 0 and self._env is not None:
+                    env = self._env.copy()
+                    env['LOKY_WORKER_ID'] = str(worker_id)
                 p = self._context.Process(target=_process_worker, args=args,
-                                          env=self._env)
+                                          env=env)
```

Comment:
I am not sure that this is the right way to pass the value using `env`.

The `env` mechanism is there to allow changing value of the `env` before any library is loaded as some value cannot be over-written afterward (`OMP_NUM_THREADS` needs to be set before loading `numpy`). This mechanism is only implemented using the `loky` backend (not `forkserver` or `spawn`).

I would rather add an argument `worker_id` to `_process_worker` and set the environment variable in the process variable so it is consistent for all workers.

Reply:
Great suggestion. I have made an attempt at implementing this.